### PR TITLE
Factor out portions of PendingClusterState bound to a specific bucket space.

### PR DESCRIPTION
**Pull request by toregge**
@vekterli, @geirst: please review

### CODE REVIEW

File: storage/src/vespa/storage/distributor/pending_bucket_space_db_transition.h [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Class used by PendingClusterState to track request bucket info
+ * reply result within a bucket space and apply it to the distributor
+ * bucket database when switching to the pending cluster state.
+ */
+class PendingBucketSpaceDbTransition : public BucketDatabase::MutableEntryProcessor
+{
+public:
+    using Entry = pendingbucketspacedbtransition::Entry;
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/pending_bucket_space_db_transition.h]: This is a bit of a mouthful 😉  Maybe we could shorten it down to e.g. `dbtransition::Entry`? Type-wise, it's really just a tuple of bucket id and bucket info for that bucket, which could theoretically be used elsewhere as well

**Review by vekterli — APPROVED** [review]
LGTM. One optional comment regarding namespace name shortening.

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**